### ISSUE TITLE
Changing the model's algorithm

### ISSUE THREAD

**Issue by Gcarmnonapy7** [Gcarmnonapy7/chatbot-ETEP#4]
On this project we are using the Bigram Language Model that predicts the next token based on the current one for a more advanced model we need to pick advanced algorithm implementation that predict tokens based on 3 or 10 currents ones to have a better result and the self attention algorithm BUT i am already working on this.
` class BigramLanguageModel(nn.Module):
  '''
  Super simple Bigram Language Model.
  Predicts the next token based only on the current one.
  '''
  def __init__(self):
     super(BigramLanguageModel,self).__init__()
     # each token reads off the logits for the next token and lookup table
     self.token_embedding = nn.Embedding(vocab,n_embed)
     self.possition_embedding = nn.Embedding(block_size,n_embed)
     self.lm_head = nn.Linear(n_embed,vocab)

  def forward(self,idx,targets=None):
    # x contains token indexs and targets are the expected next tokens. Both have shape (B, T),
    # where B = batch size, T = time steps (sequence length)
    tokn_embed = self.token_embedding(idx) # will return a tensor(b,t,c) => batch,time,channel/ bacth,tokens,tokens proba 4*8*vocab_size
    logits = self.lm_head(tokn_embed) #(B,T,C) but they are not equals

    if targets is None:
      return logits,None
    #Cross Entropy waits that the logits are in 2D
    B,T,C = logits.shape
    logits = logits.view(B*T,C) # Flattens the logits from (B, T, C) to (B*T, C) for loss computation"
    targets = targets.view(B*T)
    loss = F.cross_entropy(logits,targets)
    return logits,loss
  def generate(self,idx,max_new_tokens):
    '''
    Will generate new tokens
    '''
    # x is (B,T) array of indexs in the current context
    for _ in range(max_new_tokens):
      logits,loss = self(idx)

      logits = logits[:,-1,:]

      prob = F.softmax(logits,dim=-1)

      next_x = torch.multinomial(prob,num_samples=1)

      idx = torch.cat([idx,next_x],dim=1)
    return idx`
